src/main.ts

```diff
@@ -71,19 +71,87 @@ async function main() {
+	// - `logs`: Array of `DigestItem`s associated with the block.
+	// - `onInitialize`: Object with an array of `SanitizedEvent`s that occurred during block
+	//   initialization with the `method` and `data` for each.
+	// - `extrinsics`: Array of extrinsics (inherents and transactions) within the block. Each
+	//   contains:
+	//   - `method`: Extrinsic method, `{module}.{function}`.
+	//   - `signature`: Object with `signature` and `signer`, or `null` if unsigned.
+	//   - `nonce`: Account nonce, if applicable.
+	//   - `args`: Array of arguments.
+	//   - `tip`: Any tip added to the transaction.
+	//   - `hash`: The transaction's hash.
+	//   - `info`: `RuntimeDispatchInfo` for the transaction. Includes the `partialFee`.
+	//   - `events`: An array of `SanitizedEvent`s that occurred during extrinsic execution.
+	//   - `success`: Whether or not the extrinsic succeeded.
+	//   - `paysFee`: Whether the extrinsic requires a fee. Careful! This field relates to whether or
+	//     not the extrinsic requires a fee if called as a transaction. Block authors could insert
+	//     the extrinsic as an inherent in the block and not pay a fee. Always check that `paysFee`
+	//     is `true` and that the extrinsic is signed when reconciling old blocks.
+	// - `onFinalize`: Object with an array of `SanitizedEvent`s that occurred during block
+	//   finalization with the `method` and `data` for each.
+	//
+	// Note: Block finalization does not correspond to consensus, i.e. whether the block is in the
+	// canonical chain. It denotes the finalization of block _construction._
+	//
+	// Substrate Reference:
+	// - `DigestItem`: https://crates.parity.io/sp_runtime/enum.DigestItem.html
+	// - `RawEvent`: https://crates.parity.io/frame_system/enum.RawEvent.html
+	// - Extrinsics: https://substrate.dev/docs/en/knowledgebase/learn-substrate/extrinsics
+	// - `Extrinsic`: https://crates.parity.io/sp_runtime/traits/trait.Extrinsic.html
+	// - `OnInitialize`: https://crates.parity.io/frame_support/traits/trait.OnInitialize.html
+	// - `OnFinalize`: https://crates.parity.io/frame_support/traits/trait.OnFinalize.html
+	get('/block/', async () => {
+		const hash = await api.rpc.chain.getFinalizedHead();
 
 		return await handler.fetchBlock(hash);
 	});
 
-	get('/block/', async () => {
-		const hash = await api.rpc.chain.getFinalizedHead();
+	get('/block/:number', async (params) => {
+		const number = parseNumber(params.number);
+		const hash = await api.rpc.chain.getBlockHash(number);
 
 		return await handler.fetchBlock(hash);
 	});
 
+	// GET balance information for an address.
+	//
+	// Paths:
+	// - `address`: The address to query.
+	// - (Optional) `number`: Block number at which to query. If not provided, queries finalized head.
+	//
+	// Returns:
+	// - `at`: Block number and hash at which the call was made.
+	// - `nonce`: Account nonce.
+	// - `free`: Free balance of the account. Not equivalent to _spendable_ balance. This is the only
```

Comment:
Not necessary, but I think the docs could be more comprehensive by linking to something like this for balances: https://stackoverflow.com/questions/56418100/what-is-the-difference-between-free-balance-reserved-balance-locked-balance-v

Reply:
I don't think so, that SO isn't maintained and these should be strictly reference.